yax/shared/utilities/taxidtool.py

```diff
@@ -0,0 +1,445 @@
+    gis_to_taxids = build_gis_to_taxids(inclusion_tree)
+
+    output_sequences(taxid_data, inclusion_tree, gis_to_taxids,
+                     sequences_input_fp, output_location, project_name,
+                     truncation_level)
+    output_tree(taxid_data, inclusion_tree, output_location, project_name)
+
+
+def get_input_taxid_data(input_taxid_data_fp):
+    ##########################################################################
+    # Input
+    #   input_taxid_data_fp STRING  location of taxid_data as previously
+    #                               computed
+    #
+    # Output
+    #   taxid_data  DICT    Recreation of precomputed taxid_data structure
+    #
+    # Description
+    #   Reads in the preprocessed taxid_data file to a dictionary which the
+    #   -clip process uses.
+    ###########################################################################
+    print("Entering get_input_taxid_data")
+
+    taxid_data = {}
+    with open(input_taxid_data_fp, 'r') as taxid_data_fh:
+        counter = 0
+        for line in taxid_data_fh:
+            if counter % 10000 == 0:
+                print(counter)
+            counter += 1
+            this_line = line.rstrip().split("\t")
+            taxid = this_line[0]
+            sci_name = this_line[1]
+            assoc_gis = this_line[2].rstrip("]").lstrip("[").replace("'", "")\
+                .replace(" ", "").split(",")
+            children = list((this_line[3].rstrip("]").lstrip("[").
+                             replace("'", "").replace(" ", "").split(",")))
+            parents = list((this_line[4].rstrip("]").lstrip("[").
+                            replace("'", "").replace(" ", "").split(",")))
+            rank = this_line[5]
+
+            taxid_data[taxid] = [sci_name, assoc_gis, children, parents, rank,
+                                 0]
+
+    print("Exiting get_input_taxid_data")
+    return taxid_data
+
+
+def build_branch(taxid_data, inclusion_roots, truncation_level):
+    ##########################################################################
+    # Input
+    #   taxid_data  DICT    representation of taxid tree from NCBI
+    #   root_taxids LIST    taxids to be used as roots of tree branches
+    #
+    # Output
+    #   this_branch DICT    representation of branches built from roots
+    #
+    # Description
+    #   Responsible for building the branches from the provided roots in both
+    #   exclusion_roots and inclusion_roots. It does so by simply copying
+    #   dictionary entries from taxid_data to this_branch.
+    ###########################################################################
+    print("Entering build_branch")
+
+    this_branch = {}
+    taxids_to_add = set()
+    # inclusion_roots_check = set(inclusion_roots)
+    for taxid in inclusion_roots:
+        taxids_to_add.add(taxid)
+    while len(taxids_to_add) > 0:
+        taxids_found = set()
+        for taxid in taxids_to_add:
+            for id in taxid_data[taxid][2]:
```

Comment:
Seems to happen a lot, consider using a `namedtuple` and accessing by property if your record is predictable enough